### PR TITLE
ARCH-1046 - Changes around github packages and the npm/nuget workflows.

### PR DESCRIPTION
**Pull request by danielle-casella-adams**

### CODE REVIEW

File: workflow-templates/im-build-dotnet-ci.yml [hunk 1]
```diff
@@ -1,9 +1,13 @@
-# Workflow Code: LoathsomeSnipe_v10    DO NOT REMOVE
+# Workflow Code: LoathsomeSnipe_v11    DO NOT REMOVE
 
 # TODO: Ensure each of the repo-level and env-level secrets used in this workflow have been populated by an admin in your repository.
 
 # TODO:  If the project contains a local nuget.config remove it.  It interferes with restoring packages on the GitHub Action runners.
 
+# TODO:  If your project contains a .npmrc file that points to any of our im-* orgs as a registry you have two options:
+#        1 - Remove the .npmrc file completely and ensure you are using the im-open/authenticate-with-gh-package-registries action in each job that needs to authenticate
+#        2 - Supply a READ_PKG_TOKEN environment variable to each step that might use the .npmrc file for authentication with GH Packages (dotnet build, setup node, npm ci, etc)
```
Comment (JosephDSchwartz) [inline workflow-templates/im-build-dotnet-ci.yml:9]: They could also specify it at the job or even workflow level, right? Do we for some reason want them to do it on the individual steps?

File: workflow-templates/im-build-npm-package.yml [hunk 2]
```diff
@@ -94,14 +96,19 @@ jobs:
       #     read-pkg-token: ${{ secrets.READ_PKG_TOKEN }} # This is an org-level secret
       #     orgs: 'im-client,im-enrollment,im-practices' # TODO:  Verify list of orgs packages will be pulled from
 
+      # TODO:  If your project contains a .npmrc file that points to any of our im-* orgs as a registry you have two options:
+      #        1 - Remove the .npmrc file completely and ensure you are using the im-open/authenticate-with-gh-package-registries action in each job that needs to authenticate
+      #        2 - Supply a READ_PKG_TOKEN environment variable to each step that might use the .npmrc file for authentication with GH Packages (dotnet build, setup node, npm ci, etc)
```
Comment (JosephDSchwartz) [inline workflow-templates/im-build-npm-package.yml:101]: It feels a little off to me to have this note below a step that is using the `authenticate-with-gh-package-registries` action. Maybe if we referenced the fact that it's above like "... ensure you are using the im-open/authenticate-with-gh-package-registries action in each job that needs to authenticate, like above", or combine this note with the block above?

**Review by JosephDSchwartz — APPROVED** [review]
I have some questions, but nothing major. I'll go ahead and approve in case no action needs to be taken.

File: workflow-templates/im-build-dotnet-ci.yml (hunk 1 above)
Comment (hpractv) [reply workflow-templates/im-build-dotnet-ci.yml:9]: It does make more sense at the job or workflow particularly if it spans steps, like build and test.

File: workflow-templates/im-build-nuget-package.yml [hunk 3]
```diff
@@ -211,9 +211,16 @@ jobs:
           # GITHUB_TOKEN is a special per-job token generated by GH for interacting with the repo    
           dotnet nuget push "$nupkg" --source "https://nuget.pkg.github.com/${{ env.ORGANIZATION }}/index.json" --api-key ${{ secrets.GITHUB_TOKEN }}
 
-          # TODO:  Remove if you do not wish to dual publish to GH Packages and Artifactory
+          # TODO:  Remove the remaining items if you do not wish to publish to Artifactory
```
Comment (hpractv) [inline workflow-templates/im-build-nuget-package.yml:214]:
```suggestion
          # TODO:  Remove the remaining items if you do not wish to publish to Artifactory
```
```suggestion
          # TODO:  Remove the remaining items in this job, if you do not wish to publish to Artifactory
```

**Review by hpractv — CHANGES_REQUESTED** [review]

File: workflow-templates/im-build-npm-package.yml [hunk 4]
```diff
@@ -94,14 +96,19 @@ jobs:
...
       - name: Calculate next version
         id: version
-        uses: im-open/git-version-lite@v2.0.0
+        uses: im-open/git-version-lite@v2.0.1
         with:
           calculate-prerelease-version: ${{ env.PRERELEASE }}
           branch-name: ${{ github.head_ref }} # This is only populated when the trigger is pull_request, otherwise it is empty
-          prefix: 'v' # TODO:  verify your prefix, the new ci workflows add v automatically, if you use different things like mfe/db/etc you might need to change this
-          create-ref: ${{ env.IS_MERGE_TO_MAIN }} # TODO:  Remove if you do not want a release created for main.  If deleted, the finish-build comment on pr step should be updated since there is no release.
+          prefix: 'none' # TODO:  verify your prefix, the new ci workflows add v automatically but nuget packages typically won't have a prefix
```
[the remark this answers is not in the document]
Comment (danielle-casella-adams) [reply workflow-templates/im-build-npm-package.yml:110]: it should say npm.

File: workflow-templates/im-build-dotnet-ci.yml (hunk 1 above)
Comment (danielle-casella-adams) [reply workflow-templates/im-build-dotnet-ci.yml:9]: I'm not going to lie my mind is kind of blown right now.  For some reason I thought if you wanted to an action to have access to an env variable, (like inside the action, not in the expressions setting up the action) you had to specifically pass it to the action that way.  I thought when we were very first starting out I didn't specify it that way and the action wasn't getting the value, but there might have been something else going on.

File: workflow-templates/im-build-dotnet-ci.yml [hunk 5]
```diff
@@ -19,6 +15,7 @@ on:
     types: [opened, reopened, synchronize, closed]
 
 env:
+  READ_PKG_TOKEN: ${{ secrets.READ_PKG_TOKEN }}
```
Comment (JosephDSchwartz) [inline workflow-templates/im-build-dotnet-ci.yml:18]: Do we want to add a comment saying what level of secret (repo, org, etc.) this is, like we do for other secrets?

**Review by JosephDSchwartz — DISMISSED** [review]
One more question, but I'll approve again in case we just want to merge and potentially fix things later.

**Review by hpractv — DISMISSED** [review]
👍🏽

File: workflow-templates/im-build-dotnet-ci.yml (hunk 5 above)
Comment (danielle-casella-adams) [reply workflow-templates/im-build-dotnet-ci.yml:18]: yes. :(

**Review by hpractv — APPROVED** [review]
👍🏽

**Review by JosephDSchwartz — APPROVED** [review]

**Review by hpractv — APPROVED** [review]
👍🏽 Once more with feeling.

**Pull request merged**